### DISCUSSION TITLE
What is the format of the content of the REDIS__CONNECTION__ADDR environment variable?

### DISCUSSION THREAD

**Discussion by JasterV** [bikeshedder/deadpool#304] [Q&A] [answered]
Hi! I am using the `config` crate to deserialize the `deadpool_redis::Config` struct.

It doesn't accept this:

```
REDIS__CONNECTION__ADDR=rediss://localhost:6379
```

or this

```
REDIS__CONNECTION__ADDR__HOST=localhost
REDIS__CONNECTION__ADDR__PORT=6379
REDIS__CONNECTION__ADDR__INSECURE=false
```

I don't know what else to do, some help would be appreciated

**Comment by bikeshedder** [comment, marked as answer]
You can find the `Config` struct here: https://github.com/bikeshedder/deadpool/blob/master/redis/src/config.rs#L39

I just realized the example in the config documentation was wrong and is now fixed as of fa9c55c15cbe82bcc3371354396d19dc04ab9294.

The correct environment configuration is:

```
REDIS__URL=rediss://localhost:6379
```

The exploded `ConnectionInfo` is a bit more tricky as it involves quite a lot of nesting and is a 1:1 mapping from the underlying structures. A JSON config would look like that:

```json
{
    "redis": {
        "connection": {
            "addr": {
                "TcpTls": {
                    "host": "localhost",
                    "port": 6379,
                    "insecure": false
                }
            }
        }
    }
}
```

I'm unsure how that `addr` enum maps to the `config::Environment`. Maybe it isn't even possible with the `config` crate and the enviornment source? :thinking:

**Reply by JasterV** [reply]
using `REDIS__URL` worked, I will mark this as reply!
